ui/DesignSystem/Component/Wallet/TxListItem.svelte

```diff
@@ -0,0 +1,117 @@
+<script>
+  export let tx = null;
+  import {
+    headerIcon,
+    formatMessage,
+    formatSubject,
+    formatDate,
+    iconState,
+    iconProgress,
+    statusText,
+    subjectAvatarShape,
+    StateType,
+  } from "../../../src/transaction.ts";
+  import { microRadToRad } from "../../../src/currency.ts";
+
+  import Rad from "../Rad.svelte";
+  import { Avatar, Caption, Icon, Text, Title } from "../../Primitive";
+
+  const subject = formatSubject(tx.messages[0]);
+
+  let avatar;
+  const updateAvatar = async () => (avatar = await subject.avatarSource);
+
+  $: updateAvatar();
```

Comment:
It's a bit odd that we have to do this here. We might have to refactor the avatar fetching later on.

Reply:
copied this from the txdetail page. Up for refactoring.